applications/hotornot/src/hon_util.erl

```diff
@@ -68,32 +68,33 @@ build_keys(Number) ->
             build_keys(Rest, D, [kz_util:to_integer(D)])
     end.
 
--spec only_numeric(binary()) -> [integer()].
+-spec only_numeric(binary()) -> binary().
 only_numeric(Number) ->
     << <<N>> || <<N>> <= Number, is_numeric(N)>>.
 
 -spec is_numeric(integer()) -> boolean().
 is_numeric(N) ->
-    N >= $0 andalso N =< $9.
+    N >= $0
+        andalso N =< $9.
 
 -spec build_keys(binary(), ne_binary(), [integer()]) -> [integer()].
+build_keys(<<>>, _, Acc) -> Acc;
 build_keys(<<D:1/binary, Rest/binary>>, Prefix, Acc) ->
-    build_keys(Rest, <<Prefix/binary, D/binary>>, [kz_util:to_integer(<<Prefix/binary, D/binary>>) | Acc]);
-build_keys(<<>>, _, Acc) -> Acc.
+    build_keys(Rest, <<Prefix/binary, D/binary>>, [kz_util:to_integer(<<Prefix/binary, D/binary>>) | Acc]).
 
 %% Given a list of rates, return the list of rates whose routes regexes match the given E164
 %% Optionally include direction of the call and options from the client to match against the rate
 -spec matching_rates(kz_json:objects(), ne_binary()) ->
                             kz_json:objects().
--spec matching_rates(kz_json:objects(), ne_binary(), api_binary(), trunking_options()) ->
+-spec matching_rates(kz_json:objects(), ne_binary(), api_binary()|'undefined', ne_binary()|'undefined', trunking_options()) ->
                             kz_json:objects().
 matching_rates(Rates, DID) ->
-    matching_rates(Rates, DID, 'undefined', []).
+    matching_rates(Rates, DID, 'undefined', 'undefined', []).
 
-matching_rates(Rates, DID, Direction, RouteOptions) ->
+matching_rates(Rates, DID, Direction, ResourceId, RouteOptions) ->
     E164 = knm_converters:normalize(DID),
     [Rate || Rate <- Rates,
-             matching_rate(Rate, E164, Direction, RouteOptions)
+             matching_rate(Rate, E164, Direction, ResourceId, RouteOptions)
```

Comment:
@onnet same here. please also look into other places where you extended the funs to include ResourceId and see if you can work with an `option` in `RouteOptions`